flow/project.py

```diff
@@ -104,20 +104,63 @@
 
 class _condition(object):
 
-    def __init__(self, condition):
+    # All condition functions created using the _condition class are stored in this
+    # global registry variable, allowing direct equivalence comparison between
+    # condition objects.
+    _CONDITION_FUNCTIONS = dict()
+
+    def __init__(self, condition, _internal_lambda=False):
+        """The _internal_lambda argument is used to indicate that a condition
+        function is part of the built-in registry of lambdas and can be
+        identified as such during graph detection."""
+        conditions = type(self)._CONDITION_FUNCTIONS.setdefault(
+            'conditions', list())
```

Comment:
Searching for the string 'conditions' on this page returns nothing. When are we actually accessing this list?

Reply:
The list is never explicitly accessed outside the constructor of `_condition`. The purpose of this list is to retain a complete set of all condition functions that have been provided. The conditional on the next line serves to keep track of which user-passed conditions are lambda functions.

Reply:
So if we never actually access this dictionary with the keyword 'conditions', why is it not simply a list?